Sources/APIProvider.swift

```diff
@@ -274,3 +274,73 @@ private extension APIProvider {
     }
 
 }
+
+public struct PagedRequest<T>: AsyncSequence, AsyncIteratorProtocol where T: Decodable {
```

Comment:
I love how you adopted AsyncSequence for this! The implementation side for users look amazing due to this 💪

Reply:
Would you mind moving this into a separate file?